src/core/Akka.Cluster/ClusterHeartbeat.cs

```diff
@@ -742,12 +745,10 @@ internal class RingComparer : IComparer<UniqueAddress>
             /// <inheritdoc/>
             public int Compare(UniqueAddress x, UniqueAddress y)
             {
-                var result = Member.AddressOrdering.Compare(x.Address, y.Address);
-                if (result == 0)
-                    if (x.Uid < y.Uid) return -1;
-                    else if (x.Uid == y.Uid) return 0;
-                    else return 1;
-                return result;
+                var ha = x.Uid;
```

Comment:
Compare via node UIDs first, rather than last - less expensive and still probably unique.